src/http/io.rs

```diff
@@ -102,13 +97,15 @@ impl<T: AsyncRead + AsyncWrite> Buffered<T> {
                 Ok(n) => n,
                 Err(e) => {
                     if e.kind() == io::ErrorKind::WouldBlock {
+                        // TODO: Push this out, ideally, into http::Conn.
                         self.read_blocked = true;
```

Comment:
@seanmonstar my next CR will probably move `read_blocked` into `http::Conn::State`. I've verified that its semantically equivalent in all other locations, help me with errors.

For example:
```
    pub fn parse<S: Http1Transaction>(&mut self) -> Poll<MessageHead<S::Incoming>, ::Error> {
          loop {
              // Initially, read_buf will be empty; read_blocked=false
              match try!(S::parse(&mut self.read_buf)) {
                  Some(head) => {
                      // Can only get here if `parse` has all of the data it needed to construct a MessageHead
                      // Which means, read_buf was just written to.
                      // Which in turn means read_blocked = false; because read_from_io would have had to return Async::Ready
                      //trace!("parsed {} bytes out of {}", len, self.read_buf.len());
                     return Ok(Async::Ready(head.0))
                  },
                  None => {
                      if self.read_buf.capacity() >= MAX_BUFFER_SIZE {
                         // This error case can in theory happen when read_blocked = true or false;
                         // However, does the state still matter if we are returning an error?
                         debug!("MAX_BUFFER_SIZE reached, closing");
                         return Err(::Error::TooLarge);
                      }
                  },
              }
              // Semantically equivalent to expand the macro:
              // read_blocked = false
              // do read_from_io
              // If read_from_io == NotReady: read_blocked = true; return NotReady
              match try_ready!(self.read_from_io()) {
                 0 => {
                      trace!("parse eof");
                      //TODO: With Rust 1.14, this can be Error::from(ErrorKind)
                      return Err(io::Error::new(io::ErrorKind::UnexpectedEof, ParseEof).into());
                  }
                 _ => {},
              }
          }
      }
```

In that example, when we return an error, the connection will be closed correct? As such the state no longer matters, and I can set `read_blocked` to whatever without worry?